SubmittingPatches.rst

```diff
@@ -62,468 +66,207 @@ then you just add a line saying ::
+and why they are needed. In other words, the patch series and its description
+should be self-contained. This benefits both the patch merger(s) and reviewers.
 
-     Fixes: http://tracker.ceph.com/issues/12345
 
-is added before the ``Signed-off-by:`` line stating that this commit
-addresses http://tracker.ceph.com/issues/12345. It helps the reviewer to
-get more context of this bug, so she/he can hence update the issue on
-the bug tracker accordingly.
+Test your changes
+-----------------
 
-So a typical commit message for revising the document could look like::
+Before you open your PR, it's a good idea to run tests on your patchset. Doing
+that is simple, though the process can take a long time to complete, especially
+on older machines with less memory and spinning disks.
 
-     doc: add "--foo" option to bar
+The most simple test is to verify that your patchset builds, at least in your
+own development environment. The commands for this are::
 
-     * update the man page for bar with the newly added "--foo" option.
-     * fix a typo
+    ./install-deps.sh
+    ./do_cmake.sh
+    make
 
-     Fixes: http://tracker.ceph.com/issues/12345
-     Signed-off-by: Random J Developer <random@developer.example.org>
+Ceph comes with a battery of tests that can be run on a single machine. These
+are collectively referred to as "make check", and can be run by executing the
+following command::
 
-.. _split_changes:
+    ./run-make-check.sh
 
-4. Separate your changes
-------------------------
+If your patchset does not build, or if one or more of the "make check" tests
+fails, but the error shown is not obviously related to your patchset, don't let
+that dissuade you from opening a PR. The Ceph project has a Jenkins instance
+which will build your PR branch and run "make check" on it in a controlled
+environment.
 
-Separate *logical changes* into a single patch file.
+Once your patchset builds and passes "make check", you can run even more tests
+on it by issuing the following commands::
 
-For example, if your changes include both bug fixes and performance
-enhancements for a single driver, separate those changes into two
-or more patches. If your changes include an API update, and a new
-driver which uses that new API, separate those into two patches.
+    cd build
+    ../qa/run-standalone.sh
 
-On the other hand, if you make a single change to numerous files,
-group those changes into a single patch. Thus a single logical change
-is contained within a single patch.
+Like "make check", the standalone tests take a long time to run. They also
+produce voluminous output. If one or more of the standalone tests fails, it's
+likely the relevant part of the output will have scrolled off your screen or
+gotten swapped out of your buffer. To capture the output for later analysis,
+various tools, such as `script`_ (part of `util-linux`_ which is likely
+already installed on your machine), are available.
```

Comment:
i don't think we should include this in our document. yes, it's helpful. but it's orthogonal to Ceph development in the sense that it's not closely connected to Ceph or is not specific to Ceph. if we have this hint in this document, shall we include some links to useful Emacs packages or awesome VIM plugins? probably a link to Visual Studio Code as well?